cmake/linux/LinuxDeploy.cmake

```diff
@@ -0,0 +1,290 @@
+	"${CPACK_BINARY_DIR}/${lmms}-*.AppImage"
+	"${CPACK_BINARY_DIR}/install_manifest.txt")
+list(SORT cleanup)
+file(REMOVE ${cleanup})
+
+# Download linuxdeploy, expose bundled appimagetool to PATH
+download_binary(LINUXDEPLOY_BIN
+	"https://github.com/linuxdeploy/linuxdeploy/releases/download/continuous/linuxdeploy-${ARCH}.AppImage"
+	linuxdeploy-${ARCH}.AppImage
+	FALSE)
+
+# Symlink nested appimagetool
+set(_APPIMAGETOOL_LINK "${CPACK_CURRENT_BINARY_DIR}/appimagetool")
+if(NOT EXISTS "${_APPIMAGETOOL_LINK}")
+	set(_APPIMAGETOOL "${CPACK_CURRENT_BINARY_DIR}/.linuxdeploy-${ARCH}.AppImage/squashfs-root/plugins/linuxdeploy-plugin-appimage/appimagetool-prefix/AppRun")
+	message(STATUS "Creating a symbolic link ${_APPIMAGETOOL_LINK} which points to ${_APPIMAGETOOL}")
+	create_symlink("${_APPIMAGETOOL}" "${_APPIMAGETOOL_LINK}")
+endif()
+
+# Download linuxdeploy-plugin-qt
+download_binary(LINUXDEPLOY_PLUGIN_BIN
+	"https://github.com/linuxdeploy/linuxdeploy-plugin-qt/releases/download/continuous/linuxdeploy-plugin-qt-${ARCH}.AppImage"
+	linuxdeploy-plugin-qt-${ARCH}.AppImage
+	FALSE)
+
+message(STATUS "Creating AppDir ${APP}...")
+
+file(REMOVE_RECURSE "${CPACK_TEMPORARY_INSTALL_DIRECTORY}/include")
+file(MAKE_DIRECTORY "${APP}/usr")
+
+# Setup AppDir structure (/usr/bin, /usr/lib, /usr/share... etc)
+file(GLOB files "${CPACK_TEMPORARY_INSTALL_DIRECTORY}/*")
+list(SORT files)
+foreach(_file ${files})
+	get_filename_component(_filename "${_file}" NAME)
+	if(NOT _filename MATCHES ".AppDir")
+		file(RENAME "${_file}" "${APP}/usr/${_filename}")
+	endif()
+endforeach()
+
+# Copy Suil modules
+if(CPACK_SUIL_MODULES)
+	set(SUIL_MODULES_TARGET "${APP}/usr/lib/${CPACK_SUIL_MODULES_PREFIX}")
+	file(MAKE_DIRECTORY "${SUIL_MODULES_TARGET}")
+	file(COPY ${CPACK_SUIL_MODULES} DESTINATION "${SUIL_MODULES_TARGET}")
+endif()
+
+# Copy stk/rawwaves
+if(CPACK_STK_RAWWAVE_ROOT)
+	set(STK_RAWWAVE_TARGET "${APP}/usr/share/stk/rawwaves/")
+	file(MAKE_DIRECTORY "${STK_RAWWAVE_TARGET}")
+	file(GLOB RAWWAVES "${CPACK_STK_RAWWAVE_ROOT}/*.raw")
+	file(COPY ${RAWWAVES} DESTINATION "${STK_RAWWAVE_TARGET}")
+endif()
+
+# Ensure project's "qmake" executable is first on the PATH
+get_filename_component(QTBIN "${CPACK_QMAKE_EXECUTABLE}" DIRECTORY)
+set(ENV{PATH} "${QTBIN}:$ENV{PATH}")
+
+# Ensure "linuxdeploy-<arch>.AppImage" and "appimagetool" binaries are first on the PATH
+set(ENV{PATH} "${CPACK_CURRENT_BINARY_DIR}:$ENV{PATH}")
+
+# Promote finding our own libraries first
+set(ENV{LD_LIBRARY_PATH} "${APP}/usr/lib/${lmms}/:${APP}/usr/lib/${lmms}/optional:$ENV{LD_LIBRARY_PATH}")
+
+# Skip slow searching of copyright files https://github.com/linuxdeploy/linuxdeploy/issues/278
+set(ENV{DISABLE_COPYRIGHT_FILES_DEPLOYMENT} 1)
+
+# Patch desktop file
+file(READ "${DESKTOP_FILE}" DESKTOP_FILE_CONTENTS)
+string(REPLACE "Icon=${lmms}" "Icon=${lmms}.png" DESKTOP_FILE_CONTENTS "${DESKTOP_FILE_CONTENTS}")
+file(WRITE "${DESKTOP_FILE}" "${DESKTOP_FILE_CONTENTS}")
```

Comment:
Justification: https://github.com/LMMS/lmms/pull/7252#issuecomment-2620726599

@messmerd if this fixes your distro, please help me provide a comment explaining why we have this workaround so that we know when it can be removed.  Linux Mint shows the icon just fine in SVG form, so this may be a case of the standard changing quicker than the documentation. This theory is supported by the fact that `appimagetool` will use an SVG if found.

Reply:
> This theory is supported by the fact that appimagetool will use an SVG first if found.

This theory appears to be wrong.  I'll explain:

* After a bunch of trial and error, it appears that the `linuxdeploy` is placing `linux.svg` in there, so this assumption that an appimage prefers a vector version is true for `linuxdeploy`, false for `appimagetool`.
* appimagetool really wants the icon in the root of the appimage, it doesn't do fancy searching into `/usr/share/hicolor/...` like `linuxdeploy` does.
* This means we can simply remove the `.svg` file and the appimage should be happy.  Pushing the patch now.